src/app/controllers/MachineController.ts

```diff
@@ -451,14 +456,21 @@ export default class MachineController {
 		return Math.round((erg / length) * 10000) / 10000
 	}
 
-	protected getLastFrom<T>(dataArray: T[], callbackValue: (array: T) => any, isDiffPair: boolean = false): any {
+	/*protected getLastFromDiffPair<T>(dataArray: T[], callbackValue: (array: T) => (number | unknown)): (number | unknown) {
 		if (!dataArray || dataArray.length <= 0) return null
-		if (isDiffPair) {
+		const first = callbackValue(dataArray[dataArray.length - 1])
+		
 			if (dataArray.length <= 2) return null
-			return callbackValue(dataArray[dataArray.length - 1]) - callbackValue(dataArray[dataArray.length - 2])
-		} else {
-			return callbackValue(dataArray[dataArray.length - 1])
-		}
+			const second = callbackValue(dataArray[dataArray.length - 2])
+			if(typeof first !== 'number' || typeof second !== 'number') return null
+			return Number(first) - Number(second)
+		
```

Comment:
Personally I dislike code that is commented out within the source files. Why not remove it?

Reply:
Thanks, removed